pkg/controller/certificate-shim/helper.go

```diff
@@ -63,6 +63,45 @@ func translateAnnotations(crt *cmapi.Certificate, ingLikeAnnotations map[string]
 		crt.Spec.CommonName = commonName
 	}
 
+	if emailAddresses, found := ingLikeAnnotations[cmapi.EmailsAnnotationKey]; found {
+		crt.Spec.EmailAddresses = strings.Split(emailAddresses, ",")
+	}
+
+	subject := &cmapi.X509Subject{}
+	if organizations, found := ingLikeAnnotations[cmapi.SubjectOrganizationsAnnotationKey]; found {
+		subject.Organizations = strings.Split(organizations, ",")
+	}
+
+	if organizationalUnits, found := ingLikeAnnotations[cmapi.SubjectOrganizationalUnitsAnnotationKey]; found {
+		subject.OrganizationalUnits = strings.Split(organizationalUnits, ",")
+	}
+
+	if countries, found := ingLikeAnnotations[cmapi.SubjectCountriesAnnotationKey]; found {
+		subject.Countries = strings.Split(countries, ",")
+	}
+
+	if provinces, found := ingLikeAnnotations[cmapi.SubjectProvincesAnnotationKey]; found {
+		subject.Provinces = strings.Split(provinces, ",")
+	}
+
+	if localities, found := ingLikeAnnotations[cmapi.SubjectLocalitiesAnnotationKey]; found {
+		subject.Localities = strings.Split(localities, ",")
+	}
+
+	if postalCodes, found := ingLikeAnnotations[cmapi.SubjectPostalCodesAnnotationKey]; found {
+		subject.PostalCodes = strings.Split(postalCodes, ",")
+	}
+
+	if streetAddresses, found := ingLikeAnnotations[cmapi.SubjectStreetAddressesAnnotationKey]; found {
+		subject.StreetAddresses = strings.Split(streetAddresses, ",")
+	}
+
+	if serialNumber, found := ingLikeAnnotations[cmapi.SubjectSerialNumberAnnotationKey]; found {
+		subject.SerialNumber = serialNumber
+	}
+
+	crt.Spec.Subject = subject
```

Comment:
I agree, it's not a condition I thought of for my own use case but definitely makes sense. I didn't see anyway in the standard csv module to handle an escaped string, wasn't sure if I should try and handle the parsing of it myself (seemed dirty). Do you have any suggestions on how to elegantly handle it?

Reply:
I wouldn't have thought to use a CSV parser here but I think that's actually a great solution! I agree that writing a parser for this kind of thing feels gross and there are so many edge cases to consider, but CSV works well here, e.g.:

```go
// splitWithEscapeCSV parses the given input as a single line of CSV, which allows
// a comma-separated list of strings to be parsed while allowing commas to be present
// in each field. For example, a user can specify:
// "10 Downing Street, Westminster",Manchester
// to produce []string{"10 Downing Street, Westminster", "Manchester"}, keeping the comma
// in the first address. Empty lines or multiple CSV records are both rejected.
func splitWithEscapeCSV(in string) ([]string, error) {
    reader := csv.NewReader(strings.NewReader(in))

    records, err := reader.ReadAll()
    if err != nil {
        return nil, fmt.Errorf("failed to parse %q as CSV: %w", in, err)
    }

    if len(records) == 0 {
        return nil, fmt.Errorf("no values found after parsing %q", in)
    } else if len(records) > 1 {
        return nil, fmt.Errorf("refusing to use %q as input as it parses as multiple lines of CSV", in)
    }

    return records[0], nil
}
```

With a test string:

```go
const tst = `"1725 Slough Avenue, Suite 200, Scranton Business Park","10 Downing Street, Westminster",Something Else`
```

this produces:

```text
- 1725 Slough Avenue, Suite 200, Scranton Business Park
- 10 Downing Street, Westminster
- Something Else
```

as expected. I think that'll work nicely.

Reply:
Thanks a lot! Sorry for the long delay, I was away from the computer for most of the holidays. I've included your proposal as well as the reverse to write the csv when applying annotations on the secrets. Needs a second look though because I couldn't figure out why it was writing an empty value. Let me know :)